microprofile/WFLY-10522_microprofile-config-smallrye_extension.adoc

```diff
@@ -0,0 +1,58 @@
+= [WFLY-10522] Support Eclipse MicroProfile Config
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          microprofile,config,openshift
+
+== Overview
+
+Support Eclipse MicroProfile Config for applications deployed in WildFly.
+The integration uses the smallrye-config component to provide the MicroProfile Config implementation.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-10522[WFLY-10522]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1016[EAP7-1016]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Affected Projects or Components
+
+The implementation of Eclipse MicroProfile Config is provided by https://github.com/smallrye/smallrye-config[smallrye-config].
+
+=== Other Interested Projects
+
+== Requirements
+
+* Support Eclipse MicroProfile Config for any applications deployed in WildFly.
+* The extension is added to *all* configuration provided by WildFly (standalone and domain mode).
+
+=== Hard Requirements
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Implementation Plan
+
+* Add dependencies to `smallrye-config` and `microprofile-config-api` artifacts
+* Add a new `microprofile-config-smallrye` extension to provide Config support for deployments.
+* Add subsystem resources to enhance MicroProfile Config (e.g to support OpenShift ConfigMaps,
```

Comment:
Subsystem model needs to be described better. I find out that part of this model is described in documentation in [1], so can you please update(/finish) documentation according to my comments from [1] and refer this documentation here?
[1] https://github.com/wildfly/wildfly/pull/11320

Reply:
@marekkopecky I updated the doc in the dev PR with your suggestions